### PR TITLE
HeadOf and TailOf

### PR DESCRIPTION
**Pull request by ashtonhogan**
This is the code for this [issue](https://github.com/yegor256/cactoos/issues/185)

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#774](https://github.com/yegor256/cactoos/issues/774) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#774](https://github.com/yegor256/cactoos/issues/774) is assigned to @neonailol[/z](https://www.0crat.com/u/neonailol), here is [why](https://www.0crat.com/footprint/C63314D6Z/1062235408); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @yegor256[/z](https://www.0crat.com/u/yegor256) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer

**Comment by 0crat** [comment]
Job [#774](https://github.com/yegor256/cactoos/issues/774) is already in scope

**Comment by ashtonhogan** [comment]
I think you can merge this @neonailol - it's good to go

**Comment by ashtonhogan** [comment]
I am abandoning this PR, feel free to takeover

**Comment by 0crat** [comment]
@neonailol[/z](https://www.0crat.com/u/neonailol) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by SharplEr** [comment]
@ashtonhogan if you don't want to merge this PR, then close it. Otherwise, you have to fix problems with checks first.

File: src/main/java/org/cactoos/io/LoggingInputStream.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public long skip(final long num) throws IOException {
         this.logger.log(
             this.level.value(),
             String.format(
-                "Skipped %d byte(s) from %s.",
+                "HeadOf %d byte(s) from %s.",
```
Comment (llorllale) [inline src/main/java/org/cactoos/io/LoggingInputStream.java]: @ashtonhogan I believe this change is not required

File: src/main/java/org/cactoos/iterable/TailOf.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * HeadOf iterable.
+ *
+ * <p>There is no thread-safety guarantee.</p>
+ *
+ * @author Ashton Hogan (info@ashtonhogan.com)
+ * @version $Id$
+ * @param <T> Element type
+ * @since 0.29.3
```
Comment (llorllale) [inline src/main/java/org/cactoos/iterable/TailOf.java]: @ashtonhogan next release is `0.31`, please change accordingly

File: src/main/java/org/cactoos/iterator/Skipped.java [hunk 3]
```diff
@@ -27,7 +27,7 @@
 import java.util.NoSuchElementException;
 
 /**
- * Skipped iterator.
+ * HeadOf iterator.
```
Comment (llorllale) [inline src/main/java/org/cactoos/iterator/Skipped.java]: @ashtonhogan I believe this change is not required

File: src/test/java/org/cactoos/io/LoggingInputTest.java [hunk 4]
```diff
@@ -157,7 +157,7 @@ public void logSkipFromLargeTextFile() throws IOException {
         MatcherAssert.assertThat(
             "Can't log skip from text file",
             logger.toString(),
-            Matchers.containsString("Skipped 100 byte(s) from text file.")
+            Matchers.containsString("HeadOf 100 byte(s) from text file.")
```
Comment (llorllale) [inline src/test/java/org/cactoos/io/LoggingInputTest.java]: @ashtonhogan I believe this change is not required

File: src/test/java/org/cactoos/iterable/TailOfTest.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Test Case for {@link TailOf}.
+ * @author Ashton Hogan (info@ashtonhogan.com)
+ * @version $Id$
+ * @since 0.29.3
```
Comment (llorllale) [inline src/test/java/org/cactoos/iterable/TailOfTest.java]: @ashtonhogan next version is `0.31`, please update accordingly

File: src/test/java/org/cactoos/scalar/ReducedTest.java [hunk 6]
```diff
@@ -23,15 +23,15 @@
...
 import org.cactoos.iterable.Limited;
 import org.cactoos.iterable.RangeOf;
-import org.cactoos.iterable.Skipped;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.Matchers;
 import org.junit.Test;
 
 /**
- * Test case for {@link Skipped}.
+ * Test case for {@link HeadOf}.
```
Comment (llorllale) [inline src/test/java/org/cactoos/scalar/ReducedTest.java]: @ashtonhogan the original text had a typo here: please update to read "Test case for {@link Reduced}.".

**Comment by llorllale** [comment]
@ashtonhogan see my comments above

**Comment by SharplEr** [comment]
@llorllale this PR should not be accepted ever. The author has explained his position in #775.

**Comment by llorllale** [comment]
@SharplEr thanks

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @sharpler[/z](https://www.0crat.com/u/sharpler), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#774](https://github.com/yegor256/cactoos/issues/774) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @llorllale[/z](https://www.0crat.com/u/llorllale)

**Comment by elenavolokhova** [comment]
@0crat quality bad

**Comment by 0crat** [comment]
Quality is low, no payment, see [§31](http://www.zerocracy.com/policy.html#31)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Pull request closed without merging**